### ISSUE TITLE
Breaking on large datasets

### ISSUE THREAD

**Issue by redbmk** [dataworker/dataworker#14]
We're using Array.prototype.push.apply in two places in dw-helper when the data comes in. For really large datasets this will break. See this stack overflow question: http://stackoverflow.com/questions/22123769/rangeerror-maximum-call-stack-size-exceeded-why

We will likely want to do some performance testing to see which way would be fastest. I would imagine calling `push` 100k times would be awfully slow.

**Comment by redbmk** [comment]
I wrote up [a quick jsPerf](http://jsperf.com/dataworker-issue-14) to see what might be fastest. It seems like the best way is to keep track of the lengths of the source and destination arrays in variables, then iterate through adding to the destination array: `dest[destIndex++] = source[--sourceIndex]`. However, this doesn't take into account memory...since it's copying the entire source before `source` goes out of scope, it should need twice the amount of memory. The next fastest is 13% slower, by adding `delete source[sourceIndex]`, which should take care of the memory issue.

@FallSe7en, can you see any options I missed? I'm leaning towards the one labeled `idx->delete` for a compromise between speed and memory.

**Comment by redbmk** [comment]
Apparently with really large datasets, [adding delete slows it down very significantly](http://jsperf.com/dataworker-issue-14/2). I'm also not fully convinced that deleting from the array will actually release memory; since javascript uses garbage collection we don't actually have a whole lot of control over memory.

**Comment by redbmk** [comment]
I forgot about `pop`. I've added it to the original and replaced the link for the second one (it wouldn't let me edit so I had to create a new fork). That's much faster than `shift`, `splice`, or `delete` with large datasets and I think would have the same affect on memory. However, it's still quite a bit slower than just using indexes. If we think memory isn't a huge issue then just using the indexes seems to be the clear winner.
